src/ast/mod.rs

```diff
@@ -4829,22 +4779,169 @@ impl fmt::Display for CloseCursor {
 #[cfg_attr(feature = "visitor", derive(Visit, VisitMut))]
 pub struct Function {
     pub name: ObjectName,
-    pub args: Vec<FunctionArg>,
+    /// The arguments to the function, including any options specified within the
+    /// delimiting parentheses.
+    pub args: FunctionArguments,
     /// e.g. `x > 5` in `COUNT(x) FILTER (WHERE x > 5)`
     pub filter: Option<Box<Expr>>,
-    /// Specifies Ignore / Respect NULL within window functions.
+    /// Indicates how `NULL`s should be handled in the calculation.
+    ///
+    /// Example:
+    /// ```plaintext
+    /// FIRST_VALUE( <expr> ) [ { IGNORE | RESPECT } NULLS ] OVER ...
+    /// ```
     ///
-    /// [BigQuery](https://cloud.google.com/bigquery/docs/reference/standard-sql/navigation_functions#first_value)
     /// [Snowflake](https://docs.snowflake.com/en/sql-reference/functions/first_value)
-    pub null_treatment: Option<NullTreatmentType>,
+    pub null_treatment: Option<NullTreatment>,
+    /// The `OVER` clause, indicating a window function call.
     pub over: Option<WindowType>,
-    /// aggregate functions may specify eg `COUNT(DISTINCT x)`
-    pub distinct: bool,
```

Comment:
How do I get this property now then?

Reply:
I can see `DuplicateTreatment` now. But it can cause downstream to restructure the handling code to adjust these structural changes.

Trying it out ...

Reply:
Let us know if there is something wrong that can't be handled -- I can make another release with a fix if necessary

Reply:
I'm trying to upgrade sqlparser to 0.46.0 this morning but found that except my breaking changes on Insert and Delete struct, it includes others like this one and the JsonAccess refactor. So I stop the work until I can have time to understand the changes :P

Reply:
Let me know if there's any way I can help...

Reply:
@jmhain Thanks for offering the help. I just created https://github.com/apache/datafusion/pull/10392 that you can pull locally and run cargo build to see the compilation failure and see if you have the idea to fix it up.

Welcome to leave a comment on that PR or create a PR target to that PR. I'm actively responding.

Reply:
This is what I've got so far that I'm reasonably sure is correct without more context on DataFusion: https://github.com/apache/datafusion/commit/261f346140d0d6a265ef0f9c81ea1fcd007d5928

Out of time for now but I'll try to take a second look later tonight at the remaining JSON access changes needed